Singular/dyn_modules/systhreads/shared.cc

```diff
@@ -1479,6 +1479,7 @@ void *joinThread(ThreadState *ts) {
   ts->running = false;
   ts->active = false;
   thread_lock.unlock();
+  return result;
```

Comment:
Nothing ever seems to use the return value, so there wasn't a problem here. So an alternative fix would be to remove the return value altogether, but I reckon it is intended for a future usecase? @rbehrends ?

Reply:
This is part of the public C++ API (see the still incomplete `doc/kernel.md`), so it should indeed return `result`.